File: Java/Binary/MissingNumber.java
package com.brianprojects.LeetQues75.Binary;

import java.util.Arrays;
import java.util.HashSet;

public class MissingNumber {
    /*
    Given an array nums containing n distinct numbers in the range [0, n],
    return the only number in the range that is missing from the array.
     */

    public int missingNumber(int[] nums) {
        int res = nums.length;
        for(int i=0; i<nums.length; i++){
            res ^= i;
            res ^= nums[i];
        }
        return res;

    }
}

File: Python/Leetcode_Python/BitManipulation/MissingNumber.py
from typing import List


def missingNumber(nums: List[int]) -> int:
    n = len(nums)

    arr = [0] * (n + 1)

    for num in nums:
        arr[num] += 1

    for i in range(n + 1):
        if arr[i] == 0:
            return i

    return 0
